ckanext/canada/tests/test_contracts.py

```diff
@@ -88,26 +82,22 @@ def test_2022_fields(self):
             indigenous_business='',
         )
         with pytest.raises(ValidationError) as ve:
-            lc.action.datastore_upsert(
+            self.lc.action.datastore_upsert(
                 resource_id=self.resource_id,
                 records=[record])
-        err = ve.value.error_dict['records'][0]
-        expected = {
-            'vendor_postal_code': ['This field must not be empty'],
-            'buyer_name': ['This field must not be empty'],
-            'trade_agreement': ['This field must not be empty'],
-            'agreement_type_code': ['Discontinued as of 2022-01-01'],
-            'land_claims': ['This field must not be empty'],
-            'indigenous_business': ['This field must not be empty'],
-        }
-        assert isinstance(err, dict), err
-        for k in set(err) | set(expected):
-            assert k in err
-            assert err[k] == expected[k]
+        err = ve.value.error_dict
+        expected = ['vendor_postal_code',
+                    'buyer_name',
+                    'trade_agreement',
+                    'agreement_type_code',
+                    'land_claims',
+                    'indigenous_business']
+        assert 'records' in err
+        for k in set(err['records'][0]):
+            assert k in expected
```

Comment:
if we already have code testing the error messages themselves then let's keep it so that we know we're hitting the correct validation rules

Reply:
We could use just part of the error message and do `in` tests instead of equality if we want to make it easier to update the text of the messages, but by only checking the key we don't know that the thing we're testing is really working

Reply:
reverted simplification of error dict assertions. also updated `test_validators.py` error dict asserts to look the same for code parody.